src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -924,6 +924,10 @@ private static unsafe void EnsureInitializedCore()
 
                     // Mark us as initialized
                     s_initialized = true;
+
+                    // InitializeTerminalAndSignalHandling will reset the terminal on a normal exit.
+                    // This also resets it for termination due to an unhandled exception.
+                    AppDomain.CurrentDomain.UnhandledException += delegate { Interop.Sys.UninitializeTerminal(); };
```

Comment:
```suggestion
                    AppDomain.CurrentDomain.UnhandledException += () => { Interop.Sys.UninitializeTerminal(); };
```

Lambda syntax is preferred AFAIR.